apps/web/src/components/avatar.tsx

```diff
@@ -127,11 +115,82 @@ export function Avatar(props: AvatarProps) {
             <AvatarFallback>{getInitials(altText)}</AvatarFallback>
           </AvatarShad>
         </TooltipTrigger>
-        <TooltipContent className="flex flex-row gap-2">
-          <AvatarIcon />
+        <TooltipContent className="flex flex-col gap-1">
           {altText}
+          <AvatarIcon
+            avatarProfilePublicId={props.avatarProfilePublicId}
+            size="xs"
+            withDot
+          />
         </TooltipContent>
       </Tooltip>
     </TooltipProvider>
   );
 }
+
+const avatarIconVariants = cva('', {
+  variants: {
+    size: {
+      xs: 'text-[10px]',
+      sm: 'text-xs',
+      md: 'text-sm',
+      lg: 'text-md',
+      xl: 'text-lg'
+    }
+  },
+  defaultVariants: {
+    size: 'md'
+  }
+});
+
+type AvatarIconProps = {
+  avatarProfilePublicId: TypeId<
+    'orgMemberProfile' | 'org' | 'teams' | 'contacts'
+  >;
+  withDot?: boolean;
+} & VariantProps<typeof avatarIconVariants>;
+
+export function AvatarIcon(iconProps: AvatarIconProps) {
+  const type = inferTypeId(iconProps.avatarProfilePublicId);
+  const icon = () => {
+    switch (type) {
+      case 'orgMemberProfile':
+        return <User />;
+      case 'org':
+        return <BuildingOffice />;
+      case 'teams':
+        return <UsersThree />;
+      case 'contacts':
+        return <AddressBook />;
+      default:
+        return null;
+    }
+  };
+
+  const text = () => {
+    switch (type) {
+      case 'orgMemberProfile':
+        return 'Org Member';
+      case 'org':
+        return 'Organization';
+      case 'teams':
+        return 'Team';
+      case 'contacts':
+        return 'Contact';
+      default:
+        return null;
+    }
+  };
+
+  return (
+    <div
+      className={cn(
+        'flex flex-row items-center gap-1',
+        avatarIconVariants({ size: iconProps.size })
+      )}>
+      {iconProps.withDot && <Dot />}
+      {icon()}
+      <span className="">{text()}</span>
```

Comment:
Same as the last comment, call them like `<Icon/>` and `<Text/>` or you can also calculate the value and store them in `icon` and use it like `{icon}` not calling as function